Objects/genericaliasobject.c

```diff
@@ -626,6 +641,46 @@ static PyNumberMethods ga_as_number = {
         .nb_or = _Py_union_type_or, // Add __or__ function
 };
 
+static PyObject *
+ga_iternext(gaiterobject *gi) {
+    if (gi->obj == NULL) {
+        return NULL;
+    }
+    gaobject *alias = (gaobject *)gi->obj;
+    PyObject *starred_tuple = Py_GenericAlias(alias->origin, alias->args);
+    if (starred_tuple == NULL) {
+        return NULL;
+    }
+    ((gaobject *)starred_tuple)->starred = true;
+    Py_SETREF(gi->obj, NULL);
+    return starred_tuple;
+}
+
+static void
+ga_iter_dealloc(gaiterobject *gi) {
+    PyObject_GC_UnTrack(gi);
+    PyObject_GC_Del(gi);
+}
+
+static PyTypeObject Py_GenericAliasIterType = {
```

Comment:
This is wrong, this type needs `tp_traverse` and (possibly) `tp_clear`. Otherwise, this will crash in the GC.

Reply:
You can see the crash immediately as one of these objects is traversed (in debug mode):

```
❯ ./python
Python 3.11.0a3+ (heads/pep646-tuple:7f06e2a8dd, Feb  1 2022, 22:00:46) [GCC 11.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> iter(tuple[int])
[1]    717683 segmentation fault (core dumped)  ./python
```

Reply:
Huh, weird, I wonder why that wasn't crashing on my machine. Have implemented both `tp_traverse` and `tp_clear`, to be safe.